server/src/main/java/org/apache/cassandra/sidecar/livemigration/LiveMigrationInstanceMetadataUtil.java

```diff
@@ -0,0 +1,193 @@
+        {
+            dirsToCopy.add(instanceMetadata.localSystemDataFileDir());
+        }
+        dirsToCopy.addAll(instanceMetadata.dataDirs());
+        return dirsToCopy;
+    }
+
+    /**
+     * Returns map of directory that can be copied and the index to use while constructing
+     * url for file transfer.
+     */
+    public static Map<String, String> dirPathPrefixMap(InstanceMetadata instanceMetadata)
+    {
+        Map<String, String> dirIndexMap = new Hashtable<>();
+        dirIndexMap.put(instanceMetadata.hintsDir(), LIVE_MIGRATION_HINTS_DIR_PATH + "/0");
+        dirIndexMap.put(instanceMetadata.commitlogDir(), LIVE_MIGRATION_COMMITLOG_DIR_PATH + "/0");
+        if (instanceMetadata.savedCachesDir() != null)
+        {
+            dirIndexMap.put(instanceMetadata.savedCachesDir(), LIVE_MIGRATION_SAVED_CACHES_DIR_PATH + "/0");
+        }
+        if (instanceMetadata.cdcDir() != null)
+        {
+            dirIndexMap.put(instanceMetadata.cdcDir(), LIVE_MIGRATION_CDC_RAW_DIR_PATH + "/0");
+        }
+        if (instanceMetadata.localSystemDataFileDir() != null)
+        {
+            dirIndexMap.put(instanceMetadata.localSystemDataFileDir(), LIVE_MIGRATION_LOCAL_SYSTEM_DATA_FILE_DIR_PATH + "/0");
+        }
+        for (int i = 0; i < instanceMetadata.dataDirs().size(); i++)
+        {
+            dirIndexMap.put(instanceMetadata.dataDirs().get(i), LIVE_MIGRATION_DATA_FILE_DIR_PATH + "/" + i);
+        }
+        return dirIndexMap;
+    }
+
+    /**
+     * Returns map of directory and set of placeholders that represent the directory.
+     */
+    public static Map<String, Set<String>> dirPlaceHoldersMap(InstanceMetadata instanceMetadata)
+    {
+        Map<String, Set<String>> placeholderMap = new Hashtable<>();
+
+        placeholderMap.put(instanceMetadata.hintsDir(), Collections.singleton(HINTS_DIR_PLACEHOLDER));
+        placeholderMap.put(instanceMetadata.commitlogDir(), Collections.singleton(COMMITLOG_DIR_PLACEHOLDER));
+        if (instanceMetadata.savedCachesDir() != null)
+        {
+            placeholderMap.put(instanceMetadata.savedCachesDir(), Collections.singleton(SAVED_CACHES_DIR_PLACEHOLDER));
+        }
+
+        if (instanceMetadata.cdcDir() != null)
+        {
+            placeholderMap.put(instanceMetadata.cdcDir(), Collections.singleton(CDC_RAW_DIR_PLACEHOLDER));
+        }
+
+        if (instanceMetadata.localSystemDataFileDir() != null)
+        {
+            placeholderMap.put(instanceMetadata.localSystemDataFileDir(),
+                               Collections.singleton(LOCAL_SYSTEM_DATA_FILE_DIR_PLACEHOLDER));
+        }
+
+        List<String> dataDirs = instanceMetadata.dataDirs();
+        for (int i = 0; i < dataDirs.size(); i++)
+        {
+            String dir = dataDirs.get(i);
+            Set<String> placeholders = Collections.unmodifiableSet(
+            new HashSet<>(Arrays.asList(DATA_FILE_DIR_PLACEHOLDER, DATA_FILE_DIR_PLACEHOLDER + "_" + i)));
```

Comment:
We can simplify like this:
```suggestion
            Set<String> placeholders = Set.of(DATA_FILE_DIR_PLACEHOLDER, DATA_FILE_DIR_PLACEHOLDER + "_" + i);
```

Reply:
Thanks! Just realised that JDK 8 support is dropped for server. Done.